common/src/main/java/com/wynntils/models/character/CharacterModel.java

```diff
@@ -192,36 +190,42 @@ public void onContainerClick(ContainerClickEvent e) {
         }
     }
 
-    private void scanCharacterInfoPage() {
-        ScriptedContainerQuery query = ScriptedContainerQuery.builder("Character Info Query")
-                .onError(msg -> WynntilsMod.warn("Error querying Character Info: " + msg))
-
-                // Open compass/character menu
-                .then(QueryStep.useItemInHotbar(InventoryUtils.COMPASS_SLOT_NUM)
-                        .expectContainerTitle(ContainerModel.CHARACTER_INFO_NAME)
-                        .processIncomingContainer(this::parseCharacterContainer))
-                .build();
-
-        query.executeQuery();
-    }
-
-    private void scanSilverbullSubscriptionItem() {
-        ScriptedContainerQuery query = ScriptedContainerQuery.builder("Silverbull Subscription Query")
-                .onError(msg -> WynntilsMod.warn("Error querying Silverbull subscription: " + msg))
+    private void scanCharacterInfo() {
+        QueryBuilder queryBuilder = ScriptedContainerQuery.builder("Character Info Query");
+        queryBuilder.onError(msg -> WynntilsMod.warn("Error querying Character Info: " + msg));
+
+        // Open compass/character menu
+        queryBuilder.then(QueryStep.useItemInHotbar(InventoryUtils.COMPASS_SLOT_NUM)
+                .expectContainerTitle(ContainerModel.CHARACTER_INFO_NAME)
+                .processIncomingContainer(this::parseCharacterContainer));
+
+        if (System.currentTimeMillis() > silverbullExpiresAt.get()) {
+            // Open Cosmetics Menu
+            queryBuilder
+                    .then(QueryStep.clickOnSlot(COSMETICS_SLOT)
+                            .expectContainerTitle(ContainerModel.COSMETICS_MENU_NAME)
+                            .processIncomingContainer(this::parseCratesBombsCosmeticsContainer))
+                    .then(QueryStep.clickOnSlot(COSMETICS_BACK_SLOT)
+                            .expectContainerTitle(ContainerModel.CHARACTER_INFO_NAME));
+        } else {
+            WynntilsMod.info("Skipping silverbull subscription query ("
+                    + (silverbullExpiresAt.get() - System.currentTimeMillis()) + " ms left)");
+        }
 
-                // Open /use menu
-                .then(QueryStep.sendCommand("use")
-                        .expectContainerTitle(ContainerModel.COSMETICS_MENU_NAME)
-                        .processIncomingContainer(this::parseCratesBombsCosmeticsContainer))
-                .build();
+        // Open Guild Menu
+        queryBuilder.conditionalThen(
+                (container) -> !Models.Guild.getGuildName().isEmpty(),
```

Comment:
How would the model have a guild name, if the guild menu is yet to be parsed?

You should use the container to check for the guild item being an "active" guild item.

Reply:
Correct me if I'm wrong, but the parsing of the guild name happens already in `parseCharacterContainer()`. This predicate will get executed when the guild container should be opened, meaning after the main character container has already been parsed (and the guild item too, from which guild name and rank are obtained). Checking the guild item again, would add unnecessary complexity, no?

Reply:
Yes, that's right, actually. Can you add a short version of this is a comment? It's hard to expect any reader to know which guild related info is parsed here and which is parsed before.